### DISCUSSION TITLE
Feature Request: Change StringView to be compatible with the newly added Arrow StringView

### DISCUSSION THREAD

**Discussion by alamb** [duckdb/duckdb#9248] [Ideas]
(👋  Apache Arrow PMC Chair here)

# Background

DuckDB has posted about their being a zero copy  integration with Arrow: https://duckdb.org/2021/12/03/duck-arrow.html, however there are some details of transferring variable length string data that is not zero copy due to differences in how such data is represented. 

Recently [Apache Arrow](https://arrow.apache.org) added [`StringView`](https://github.com/apache/arrow/blob/9d6d501/docs/source/format/Columnar.rst#L369-L379)  to the official spec (https://github.com/apache/arrow/issues/35627) and one of the primary goals was "Zero Copy Compatibility with DuckDB"

However,  we have hit snag recently in that existing implementation of variable length strings in DuckDB is not compatible with what is (now) in the Arrow Spec.  Specifically, directly using pointers (rather than a base + offset) doesn't work for many language implementations. 

# Request:
1. Please provide your input in the current discussion thread: https://lists.apache.org/thread/l8t1vj5x1wdf75mdw3wfjvnxrfy5xomy about if you would even consider this request
2. Please consider updating DuckDB to use the adopted Arrow `StringView` spec 

p.s. I am sorry if there is an existing issue / discussion about this. I didn't find any existing discussion of this feature in the duckdb issues:
* https://github.com/duckdb/duckdb/issues?q=is%3Aissue+stringview+arrow
* https://github.com/duckdb/duckdb/discussions?discussions_q=arrow+stringview

**Comment by pdet** [comment]
Hi @alamb 

We sent an email with our POV to the [discussion list](https://lists.apache.org/thread/l8t1vj5x1wdf75mdw3wfjvnxrfy5xomy) earlier today.

Maybe it still needs some internal approval to go through?

**Reply by ianmcook** [reply]
@pdet it looks like it didn't go through. I don't think there is any pre-moderation/internal approval mechanism, so you might need to try again. Did you send it from an address that is subscribed to the `dev@` mailing list ([instructions here](https://arrow.apache.org/community/))?

**Reply by pdet** [reply]
@ianmcook, we might have missed that! Thanks for the pointer. We will try again! :)

**Reply by ianmcook** [reply]
@pdet after subscribing, if you re-send the email to dev@arrow.apache.org it with the subject line:

Re: [DISCUSS][C++] Raw pointer string views

then I think it should end up in the right place :)

**Reply by alamb** [reply]
Thank you @pdet  and @Mytherin for your input. Response is [here](https://lists.apache.org/thread/3qhkomvvc69v3gkotbwldyko7yk9cs9k) for anyone else looking for it

**Comment by jhorstmann** [comment]
Supposedly support for arrow string views was implemented in https://github.com/duckdb/duckdb/pull/10481

**Comment by alamb** [comment]
Sounds like this has been done per @jhorstmann 's comment in https://github.com/duckdb/duckdb/discussions/9248#discussioncomment-11213465. Thank you
